sky/server/server.py

```diff
@@ -1756,57 +1758,119 @@ async def kubernetes_pod_ssh_proxy(websocket: fastapi.WebSocket,
         pid=os.getpid())
     ssh_failed = False
     websocket_closed = False
+    
     try:
         conn_gauge.inc()
-        # Connect to the local port
-        reader, writer = await asyncio.open_connection('127.0.0.1', local_port)
-
-        async def websocket_to_ssh():
+        # Create a blocking socket connection to the SSH port
+        ssh_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        ssh_socket.connect(('127.0.0.1', local_port))
+        
+        # Get the current event loop for thread-safe WebSocket operations
+        loop = asyncio.get_running_loop()
+        
+        # Shared state for coordinating shutdown
+        shutdown_event = threading.Event()
+        
+        def websocket_to_ssh_thread():
+            """Thread function: WebSocket -> SSH (blocking)"""
+            nonlocal ssh_failed, websocket_closed
             try:
-                async for message in websocket.iter_bytes():
-                    writer.write(message)
+                while not shutdown_event.is_set():
                     try:
-                        await writer.drain()
-                    except Exception as e:  # pylint: disable=broad-except
-                        # Typically we will not reach here, if the ssh to pod
-                        # is disconnected, ssh_to_websocket will exit first.
-                        # But just in case.
-                        logger.error('Failed to write to pod through '
-                                     f'port-forward connection: {e}')
-                        nonlocal ssh_failed
-                        ssh_failed = True
+                        # Get data from websocket (this needs to be done in the main loop)
+                        future = asyncio.run_coroutine_threadsafe(
+                            websocket.receive_bytes(), loop)
```

Comment:
If we use the main loop anyway, why would a separate thread help then? It will still be blocked by laggy asnycio event loop right?